libs/market-depth/src/lib/orderbook.tsx

```diff
@@ -477,157 +61,119 @@ export const Orderbook = ({
+  const tableBodyUp = askRows?.length ? (
     <div
-      className={classNames(
-        'grid grid-cols-4 gap-1 text-right auto-rows-[17px]',
-        isContinuousMode ? 'grid-cols-3' : 'grid-cols-4'
-      )}
+      className="text-right w-full flex flex-col justify-end"
+      style={{ height: 'calc(50% - 20px)' }}
     >
-      {data.map((data, i) => (
-        <OrderBookRowComponent
-          key={data.price}
-          price={(BigInt(data.price) / BigInt(resolution)).toString()}
-          onClick={onClick}
-          decimalPlaces={decimalPlaces - Math.log10(resolution)}
-          positionDecimalPlaces={positionDecimalPlaces}
-          bid={data.bid}
-          relativeBid={data.relativeBid}
-          cumulativeBid={data.cumulativeVol.bid}
-          cumulativeRelativeBid={data.cumulativeVol.relativeBid}
-          ask={data.ask}
-          relativeAsk={data.relativeAsk}
-          cumulativeAsk={data.cumulativeVol.ask}
-          cumulativeRelativeAsk={data.cumulativeVol.relativeAsk}
-          indicativeVolume={
-            marketTradingMode !==
-              Schema.MarketTradingMode.TRADING_MODE_CONTINUOUS &&
-            indicativePrice === data.price
-              ? indicativeVolume
-              : undefined
-          }
-        />
-      ))}
+      <div className="grid auto-rows-[17px] gap-1">
+        {askRows.map((data, i) => (
+          <OrderbookRow
+            key={data.price}
+            price={(BigInt(data.price) / BigInt(resolution)).toString()}
+            onClick={onClick}
+            decimalPlaces={decimalPlaces - Math.log10(resolution)}
+            positionDecimalPlaces={positionDecimalPlaces}
+            bid={data.bid}
+            relativeBid={data.relativeBid}
+            cumulativeBid={data.cumulativeVol.bid}
+            cumulativeRelativeBid={data.cumulativeVol.relativeBid}
+            ask={data.ask}
+            relativeAsk={data.relativeAsk}
+            cumulativeAsk={data.cumulativeVol.ask}
+            cumulativeRelativeAsk={data.cumulativeVol.relativeAsk}
+            indicativeVolume={
+              marketTradingMode !==
+                Schema.MarketTradingMode.TRADING_MODE_CONTINUOUS &&
+              indicativePrice === data.price
+                ? indicativeVolume
+                : undefined
+            }
+            type={VolumeType.ask}
+          />
+        ))}
+      </div>
     </div>
   ) : null;
 
-  const c = theme === 'dark' ? colors.neutral[600] : colors.neutral[300];
-  const gradientStyles = isContinuousMode
-    ? `linear-gradient(${c},${c}) 33.4% 0/1px 100% no-repeat, linear-gradient(${c},${c}) 66.7% 0/1px 100% no-repeat`
-    : `linear-gradient(${c},${c}) 25% 0/1px 100% no-repeat, linear-gradient(${c},${c}) 50% 0/1px 100% no-repeat, linear-gradient(${c},${c}) 75% 0/1px 100% no-repeat`;
+  const tableBodyDown = bidRows?.length ? (
+    <div
+      className="text-right w-full flex flex-col justify-start"
+      style={{ height: 'calc(50% - 20px)' }}
```

Comment:
Can we just use AutoSizer to give us the height? We'll end up with less hard coded values for layout. We can then also remove the useResizeObserver callback logic.

Reply:
we can't use it since we have rich memoization in hooks. We would need to completely refactor the solution for it.